pkgs/development/libraries/opencv/4.x.nix

```diff
@@ -300,6 +308,11 @@ effectiveStdenv.mkDerivation {
   patches =
     [
       ./cmake-don-t-use-OpenCVFindOpenEXR.patch
+      # remove patch after opencv v4.12.0
+      (fetchpatch2 {
+        url = "https://github.com/opencv/opencv/pull/27428.patch?full_index=1";
```

Comment:
We can't fetch PRs directly, they may change. Since the PR was merged we can use `https://github.com/opencv/opencv/commit/bbe2f50b5d50a282b2260100be9e559067e55fbf.patch?full_index=1` instead

Reply:
I believe we can use them only if they are merged, because then they won't change. only unmerged pr links aren't allowed. as per what I read here https://github.com/NixOS/nixpkgs/pull/405372#issuecomment-2869177801

Reply:
Yes, the PR was merged (you can check on the commit view, it shows that the commit is present in the "4.x" branch). For unmerged PRs we vendor the patch file in nixpkgs.

Reply:
I have seen other committers require using merged commits instead of merged PRs. Though you are right that the patch file wouldn't change at this point.

Reply:
changed it to commit